lib/Sema/CSDiag.cpp

```diff
@@ -2743,11 +2743,24 @@ diagnoseUnviableLookupResults(MemberLookupResult &result, Type baseObjTy,
           return;
         }
       }
-        
+
+      // Check whether the instance member is declared on parent context and if so
+      // provide more specialized message
+      auto memberTypeContext = member->getDeclContext()->getInnermostTypeContext();
+      auto currentTypeContext = CS.DC->getInnermostTypeContext();
+      if (memberTypeContext->getSyntacticDepth() < currentTypeContext->getSyntacticDepth()) {
+        diagnose(loc, diag::could_not_use_instance_member_of_outer_type_on_nested_type,
```

Comment:
I'd rather not have a new diagnostic here. Can we emit the existing diagnostic, and have the fixit fully qualify the type name instead?

Reply:
Thanks @slavapestov!
Could you explain the reason? I'd like to understand pros and cons of doing this. 
Do I understand correctly that you suggest in this case:
```
struct Outer {
  var outer: Int

  struct Inner {
    var inner: Int

    func sum() -> Int {
      return inner + outer
    }
  }
}
```
emit error:
```
instance member 'outer' cannot be used on type 'Outer.Inner'
```

instead of current:
```
instance member 'outer' cannot be used on type 'Outer'
```
Also: what do you mean by fixit? I think that compiler doesn't generate fixit in this case, at least I don't see it in Xcode.

Thanks for the help!

Reply:
I think what @slavapestov is trying to say is that we should fix the argument of the existing diagnostic `diag::could_not_use_instance_member_on_type` to say 'Outer.Inner' vs. only 'Outer' right now (first argument `instanceType`) instead of creating new diagnostic with the same text, because it's already reported right after special cased one.

Reply:
@xedin Yeah, that’s understandable but new diagnostic doesn’t have the same text. It has different one, with different number of template arguments(3 in this case). With new one there is following message: “instance member 'outer' of type 'Outer' cannot be accessed on instance of nested type 'Outer.Inner”. This gives more info than “instance member ‘outer’ cannot be used on type ‘Outer.Inner’(about the variable definition scope). If we want to have this(like specified in JIRA ticket) we need to have a new diagnostic. If it’s not necessary I could of course change it and implement it as @slavapestov suggests. What do you think?

Reply:
@Kacper20 There is a way to make existing one to distinguish between `on type` and `on nested type`, you can add `on %select{type|nested type}2` and pass the flag as a third argument (please take a look at how other diagnostics use it in `DiagnosticsSema.def`).

Reply:
Thanks @xedin. I believe the same could be done with `of type ‘Outer’`, which is also additional? Don’t know how clear will it be but let’s find out :) That’s a great tip, will look at it.